### ISSUE TITLE
input-date-picker does not announce the purpose of each input when in range mode

### ISSUE THREAD

**Issue by sansth1010** [Esri/calcite-design-system#7993]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

US Department of Veterans Affairs has reported accessibility bug on their Hub site's [search page ](https://vetdata-va-ncvas.hub.arcgis.com/explore?layout=list) which uses `input-date-picker` component to select date range. `input-date-picker` doesn't announce what each field in the component is for.

### Actual Behavior

As seen in the video below, when input field of `input-date-picker` gets in focus - voice over says "You are currently in a menu pop-up combo box".  Please unmute the video to hear the voice over. You can also see texts of what voice over is saying on the bottom left screen.

https://github.com/Esri/calcite-design-system/assets/94729944/8aa810d3-4f1e-4a2f-8dc4-fb0f0304dcfa

### Expected Behavior

We should hear labels associated with the inputs inside of `input-date-picker` range like "Start date" and "End date".

### Reproduction Sample

https://codepen.io/sansth1010/pen/rNobOwe

### Reproduction Steps

1. Open VoiceOver (CMD+F5)
2. Click on "Date range" to expand accordion.
3. Press Ctrl+Option+right-arrow to move focus to first date range input

### Reproduction Version

1.4.1

### Priority impact

p2 - want for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
Related: https://github.com/Esri/calcite-design-system/issues/8480

**Comment by driskull** [comment]
@geospatialem part of the issue might be that the assistive text element is set to `aria-hidden="true"`

https://github.com/Esri/calcite-design-system/blob/9bfbb42686fbbcc08d5ffeafaba7fab2d4199df8/packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx#L591-L593

I'm wondering if this should just use the `label` property of the `calcite-input-text` instead of `aria-describedby`? Seems like an label is more appropriate here?

Also the `Date Format:` text isn't localized. I think we need a string `Date Format: {format}`

@macandcheese @anveshmekala looks like `table-cell`, `table-header`, and `text-area` are doing the same thing. I don't think these elements made for screen readers should have `aria-hidden` on them.

https://github.com/Esri/calcite-design-system/blob/9bfbb42686fbbcc08d5ffeafaba7fab2d4199df8/packages/calcite-components/src/components/table-cell/table-cell.tsx#L247-L250

https://github.com/Esri/calcite-design-system/blob/9bfbb42686fbbcc08d5ffeafaba7fab2d4199df8/packages/calcite-components/src/components/table-header/table-header.tsx#L264-L267

https://github.com/Esri/calcite-design-system/blob/9bfbb42686fbbcc08d5ffeafaba7fab2d4199df8/packages/calcite-components/src/components/text-area/text-area.tsx#L369-L371

**Comment by driskull** [comment]
the `aria-hidden` on those elements isn't using `toAriaBoolean` so they really aren't doing anything anyway.

**Comment by macandcheese** [comment]
Yeah happy to update that as needed - For the table-cell it was an attempt [at a Safari specific workaround] (https://github.com/Esri/calcite-design-system/blob/9bfbb42686fbbcc08d5ffeafaba7fab2d4199df8/packages/calcite-components/src/components/table/table.tsx#L170) we can remove for the next major browser version, I think.

**Comment by driskull** [comment]
I wonder if `input-date-picker` should have its own `label` property for users to set a custom label?

Currently, they are stuck with `Date Format`. If they have multiple date pickers this wouldn't be very useful.

**Comment by geospatialem** [comment]
> I wonder if `input-date-picker` should have its own `label` property for users to set a custom label?
> 
> Currently, they are stuck with `Date Format`. If they have multiple date pickers this wouldn't be very useful.

@driskull I like this approach. This also came up in a Living Atlas audit in December 2024. Having a `label` attribute with more context would be beneficial.

Currently in NVDA the following is read back when the component does not have a value, which could be improved upon:

> MM/DD/YYYY edit blank

**Comment by geospatialem** [comment]
Verified with `3.3.0-next.13` for both `range` and non-range configurations with NVDA and JAWS.
